lib/Distribution/Server/Util/CabalRevisions.hs

```diff
@@ -442,28 +454,30 @@ checkDependencies componentName ds1 ds2 = do
 
     depKind = depTypeName dproxy ++ " dependency"
 
-class (Ord (DepKey d)) => IsDependency d where
+class (Ord (DepKey d), Pretty vr, Eq vr) => IsDependency vr d | d -> vr where
```

Comment:
Out of curiosity, why a functional dependency here where `DepKey` is an associated type?